engine/src/ui/canvas/plugin.cpp

```diff
@@ -30,20 +63,36 @@ void cubos::engine::uiCanvasPlugin(Cubos& cubos)
     cubos.component<UICanvas>();
     cubos.component<UIHorizontalStretch>();
     cubos.component<UIVerticalStretch>();
+    cubos.uninitResource<State>();
 
-    cubos.tag(elementVPUpdateTag);
-    cubos.tag(elementPropagateTag).after(elementVPUpdateTag);
-    cubos.tag(uiBeginTag).after(elementPropagateTag);
-    cubos.tag(uiDrawTag).after(uiBeginTag).tagged(drawToRenderTargetTag);
+    cubos.tag(uiElementVPUpdateTag);
+    cubos.tag(uiElementPropagateTag).after(uiElementVPUpdateTag);
+    cubos.tag(uiBeginTag).after(uiElementPropagateTag);
+    cubos.tag(uiDrawTag).after(uiBeginTag);
+    cubos.tag(uiEndTag).after(uiDrawTag).tagged(drawToRenderTargetTag);
 
     cubos.observer("initialize Canvas").onAdd<UICanvas>().call([](Query<UICanvas&> query) {
         for (auto [canvas] : query)
         {
             canvas.mat = glm::ortho<float>(0, canvas.referenceSize.x, 0, canvas.referenceSize.y);
         }
     });
+
+    cubos.startupSystem("setup canvas state")
+        .after(windowInitTag)
+        .call([](Commands cmds, const core::io::Window& window) {
+            cubos::core::gl::DepthStencilStateDesc dssd;
+            dssd.depth.enabled = true;
+            dssd.depth.writeEnabled = true;
+            dssd.depth.near = -1.0F;
+            dssd.depth.compare = Compare::LEqual;
+            cmds.emplaceResource<State>(window->renderDevice().createDepthStencilState(dssd),
+                                        window->renderDevice().createConstantBuffer(sizeof(glm::mat4), nullptr,
+                                                                                    cubos::core::gl::Usage::Dynamic));
+        });
+
     cubos.system("set element matrix")
-        .tagged(elementVPUpdateTag)
+        .tagged(uiElementVPUpdateTag)
         .with<UIElement>()
```

Comment:
There's no longer a VP matrix in UI element, this system should be renamed.
The same applies to the tag.